spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/TraceFilterTests.java

```diff
@@ -158,8 +158,10 @@ public void startsNewTraceWithParentIdInHeaders() throws Exception {
 
 		filter.doFilter(this.request, this.response, this.filterChain);
 
-		// this creates a child span which is why we'd expect the parents to include 1L)
-		assertThat(this.span.getParents()).containsOnly(3L);
+		// this creates a child span which is why we'd expect the parents to include the parent id
+		// especially important if no handler interceptors have been used. Then we need to have
+		// an additional span created on the server side so the call graph in Zipkin looks good
```

Comment:
"so the call graph in Zipkin looks good" is something that can be pinned down a bit better. The harder-hitting code comment could be "We add a child span on the server side to show which controller serviced the request, and for how long"